### ISSUE TITLE
Monitoring records grad_norm 0.0 when none was logged, and carries the last value into steps that logged none

### ISSUE THREAD

**Issue by Srinivasan8888** [MakazhanAlpamys/Soup#1147]
Filed as offered on #836: the maintainer ruled there that `monitoring/callback.py:115`'s `0.0` default "is a defect there, not a convention to copy" and asked for it as its own issue.

## What happens

`SoupTrainerCallback.__init__` sets `self._last_grad_norm = 0.0` (`callback.py:115`). `on_log` overwrites it only when the log carries `grad_norm` (`:248-249`) and hands it to every sink: the display, the SSE stream (`:309`) and the tracker (`:392`). That writes two kinds of value that were never measured:

1. **A fabricated `0.0`** for every step before the backend first logs a norm, and for every step of a backend that never logs one (MLX leaves it unset by design).
2. **A carried value** on any step whose log has no norm: the last real one is re-recorded.

Measured on `main` `58c2e3cf`, driving the real callback with a tracker double:

```
step 1  logs: loss, lr                 -> tracker grad_norm = 0.0   (nothing measured)
step 2  logs: loss, lr, grad_norm=0.7  -> tracker grad_norm = 0.7
step 3  logs: loss, lr                 -> tracker grad_norm = 0.7   (nothing measured)
recorded: [0.0, 0.7, 0.7]
```

## Why it matters

- A stored `0.0` is the signature of a run that was not training. #836's contract exists to catch exactly that, and here "absent" and "genuinely zero" are the same value in the run database.
- The same file already refuses both mistakes for validation loss: `_last_val_loss` starts at `None` (`:118`). The comment at `:260-272` explains why the **record** must not carry a value forward ("That is the same fabrication this change refuses..."). `grad_norm` gets neither treatment.
- `experiment/tracker.py:348` defaults `grad_norm: float = 0.0` as well, so fixing only the callback would still let a caller omit it and store zero.

## Suggested shape (not implemented; I'm not claiming this)

- Start at `None`.
- The panel may carry the last value forward, as `val_loss` does.
- The record passes only the norm in *this* log (`logs.get("grad_norm")`) and stores `NULL` otherwise.
- `tracker.log_metrics(grad_norm=None)` default.
- `display.py:88`/`:129` already hide a non-positive norm, so the panel needs care to show a real `0.0` rather than hide it.

## Acceptance

- [ ] A run whose logs never carry `grad_norm` stores `NULL`, not `0.0`, for every step.
- [ ] A step whose log has no norm stores `NULL`, not the previous step's value.
- [ ] A logged `0.0` is stored as `0.0` (the control: a real zero must survive).
- [ ] Readers of the series (`soup runs`, `soup why`, the Web UI) handle `NULL` without crashing. Each consumer is a test.

**Comment by taljeon** [comment]
I’ll take this as the separate monitoring follow-up requested on #836.

Plan: start the callback’s last grad norm at `None`, keep the last measured value only for the live panel, and send only the norm in the current log to SSE and the experiment tracker. Change `ExperimentTracker.log_metrics`’s omitted default to `None` so SQLite stores `NULL`, while an explicitly logged `0.0` stays a real zero. Make the terminal panel render a real zero without inventing one before any measurement, and keep missing Web UI chart points as `null` rather than coercing them to zero. I’ll add focused tests for the three-step absent/0.7/absent sequence plus a real-zero control, and verify `soup runs`, `soup why`, and the Web UI metric/chart path tolerate `NULL` without fabricating a reading. The existing #1146 benchmark PR touches a separate set of files.

AI disclosure: I’m using OpenAI Codex for source analysis, implementation, testing, and publication preflight. The registered local Qwen candidate route remains stopped before inference by the scheduled-configuration registration check; I will not reset or bypass it, and no Qwen output is used. I’ll review the exact diff and use GitHub noreply commit metadata before publishing.

**Comment by MakazhanAlpamys** [comment]
Fixed in `main` by @taljeon (#1149). An absent gradient norm now stays absent at every sink — the SQLite column stores NULL, the SSE payload omits the key, the terminal panel shows no Grad line, and the Web UI chart draws a gap rather than a point at zero. A **measured** 0.0 is now visible where it used to be hidden by the `> 0` gate.

Measured over `[absent, 0.7, absent, 0.0]`: on main the run database held `[0.0, 0.7, 0.7, 0.0]` with `typeof()` real at every row; it now holds `[None, 0.7, None, 0.0]`.

No migration is needed — the column was already nullable. `backend: mlx` and `task: ppo` inherit the fix for free, since both already call `log_metrics` without a norm.

Try: `soup runs show <run>` or `soup why <run>` on a run from a backend that does not log one.
